tst/EnergyPlus/unit/ZoneContaminantPredictorCorrector.unit.cc

```diff
@@ -390,3 +390,210 @@ TEST_F(EnergyPlusFixture, ZoneContaminantPredictorCorrector_CorrectZoneContamina
+    Zone(3).ZoneEqNum = 1;
+    Zone(3).Multiplier = 1.0;
+    Zone(3).Volume = 1000.0;
+    Zone(3).SystemZoneNodeNumber = 5;
+    Zone(3).ZoneVolCapMultpMoist = 1.0;
+
+    OutBaroPress = 101325.0;
+
+    NumZoneReturnPlenums = 0;
+    NumZoneSupplyPlenums = 0;
+
+    OAMFL.allocate(3);
+    VAMFL.allocate(3);
+    EAMFL.allocate(3);
+    CTMFL.allocate(3);
+    MDotOA.allocate(3);
+    MDotOA = 0.001;
+    ScheduleManager::Schedule.allocate(1);
+
+    ScheduleManager::Schedule(1).CurrentValue = 1.0;
+
+    SimulateAirflowNetwork = 0;
+
+    ZoneAirSolutionAlgo = UseEulerMethod;
+
+    Node(1).MassFlowRate = 0.01; // Zone inlet node 1
+    Node(1).HumRat = 0.008;
+    Node(2).MassFlowRate = 0.02; // Zone inlet node 2
+    Node(2).HumRat = 0.008;
+    ZoneEquipConfig(1).ZoneExhBalanced = 0.0;
+    Node(3).MassFlowRate = 0.00; // Zone exhaust node 1
+    ZoneEquipConfig(1).ZoneExh = Node(3).MassFlowRate;
+    Node(3).HumRat = 0.008;
+    Node(4).MassFlowRate = 0.03; // Zone return node
+    Node(4).HumRat = 0.000;
+    Node(5).HumRat = 0.000;
+    OAMFL(1) = 0.0;
+    VAMFL(1) = 0.0;
+    EAMFL(1) = 0.0;
+    CTMFL(1) = 0.0;
+    OAMFL(2) = 0.0;
+    VAMFL(2) = 0.0;
+    EAMFL(2) = 0.0;
+    CTMFL(2) = 0.0;
+    OAMFL(3) = 0.0;
+    VAMFL(3) = 0.0;
+    EAMFL(3) = 0.0;
+    CTMFL(3) = 0.0;
+    ZoneAirHumRat(1) = 0.008;
+    ZT(1) = 24.0;
+    ZoneAirHumRat(2) = 0.008;
+    ZT(2) = 23.5;
+    ZoneAirHumRat(3) = 0.008;
+    ZT(3) = 24.5;
+    MixingMassFlowZone = 0.0;
+
+    Node(6).MassFlowRate = 0.01;
+    Node(7).MassFlowRate = 0.01;
+    Node(8).MassFlowRate = 0.01;
+    Node(9).MassFlowRate = 0.01;
+
+    DataContaminantBalance::GCPredictedRate.allocate(3);
+    DataContaminantBalance::ZoneSysContDemand.allocate(3);
+    DataContaminantBalance::NumContControlledZones = 3;
+
+    DataContaminantBalance::ContaminantControlledZone.allocate(3);
+
+    DataContaminantBalance::ContaminantControlledZone(1).AvaiSchedPtr = 1;
+    DataContaminantBalance::ContaminantControlledZone(1).ActualZoneNum = 1;
+    DataContaminantBalance::ContaminantControlledZone(1).NumOfZones = 1;
+    DataContaminantBalance::ContaminantControlledZone(2).AvaiSchedPtr = 1;
+    DataContaminantBalance::ContaminantControlledZone(2).ActualZoneNum = 2;
+    DataContaminantBalance::ContaminantControlledZone(2).NumOfZones = 1;
+    DataContaminantBalance::ContaminantControlledZone(3).AvaiSchedPtr = 1;
+    DataContaminantBalance::ContaminantControlledZone(3).ActualZoneNum = 3;
+    DataContaminantBalance::ContaminantControlledZone(3).NumOfZones = 1;
+
+    PredictZoneContaminants(ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep);
+    EXPECT_NEAR(19.549478386, DataContaminantBalance::GCPredictedRate(1), 0.00001);
+    EXPECT_NEAR(20.887992514, DataContaminantBalance::GCPredictedRate(2), 0.00001);
+    EXPECT_NEAR(21.251538064, DataContaminantBalance::GCPredictedRate(3), 0.00001);
+
+    DataContaminantBalance::Contaminant.CO2Simulation = false;
```

Comment:
Same comment here about moving this "reset" type statements into the clear_state function in DataContaminantBalance.cc.

Reply:
@Myoldmopar I am going to move these reset statement into the clear_state function.